src/ucp/core/ucp_ep.c

```diff
@@ -219,6 +207,225 @@ ucp_ep_adjust_params(ucp_ep_h ep, const ucp_ep_params_t *params)
     return UCS_OK;
 }
 
+ucs_status_t ucp_ep_add_connected_lane(ucp_ep_h ucp_ep, uct_ep_h uct_ep)
+{
+    ucp_ep_config_key_t new_key;
+
+    if (ucp_ep_get_connected_lane(ucp_ep) != UCP_NULL_LANE) {
+        return UCS_ERR_ALREADY_EXISTS;
+    }
+
+    ucs_assert(ucp_ep_get_connected_ep(ucp_ep) == NULL);
+
+    new_key = ucp_ep_config(ucp_ep)->key;
+    if (new_key.num_lanes == UCP_MAX_LANES) {
+        return UCS_ERR_EXCEEDS_LIMIT;
+    }
+
+    new_key.connected_lane                  = new_key.num_lanes++;
+    ucp_ep->uct_eps[new_key.connected_lane] = uct_ep;
+    return ucp_worker_get_ep_config(ucp_ep->worker, &new_key, 0,
```

Comment:
must be careful changing cfg_index of an ep which can already have operations in progress